gallery_dl/extractor/instagram.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018 Leonardo Taccari
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extract images from Instagram"""
+
+import hashlib
+import json
+from .common import Extractor, Message
+from .. import text
+
+
+class InstagramExtractor(Extractor):
+    """Base class for instagram extractors"""
+    category = "instagram"
+    directory_fmt = ["{category}", "{username}"]
+    filename_fmt = "{media_id}.{extension}"
+    archive_fmt = "{media_id}"
+    root = "https://www.instagram.com"
+
+    def __init__(self, match):
+        Extractor.__init__(self)
+
+    def items(self):
+        yield Message.Version, 1
+
+        for data in self.instagrams():
+            yield Message.Directory, data
+
+            if data['typename'] == 'GraphImage':
+                yield Message.Url, data['display_url'], \
+                    text.nameext_from_url(data['display_url'], data)
+            elif data['typename'] == 'GraphSidecar':
+                # TODO: Extract all images in edge_sidecar_to_children
+                # TODO: instead of just extracting the main one!
+                yield Message.Url, data['display_url'], \
+                    text.nameext_from_url(data['display_url'], data)
+            elif data['typename'] == 'GraphVideo':
+                yield Message.Url, \
+                    'ytdl:{}/p/{}/'.format(self.root, data['shortcode']), data
+
+    def _extract_shared_data(self, page):
+        return json.loads(text.extract(page,
+                          'window._sharedData = ', ';</script>')[0])
+
+    def _extract_postpage(self, url):
+        page = self.request(url).text
+        shared_data = self._extract_shared_data(page)
+        media = shared_data['entry_data']['PostPage'][0]['graphql']['shortcode_media']
+
+        return {
+            'media_id': media['id'],
+            'shortcode': media['shortcode'],
+            'typename': media['__typename'],
+            'display_url': media['display_url'],
+            'height': int(media['dimensions']['height']),
+            'width': int(media['dimensions']['width']),
+            'comments': int(media['edge_media_to_comment']['count']),
+            'likes': int(media['edge_media_preview_like']['count']),
```

Comment:
Use `text.parse_int()` to convert numbers to integer. It doesn't throw an exception if things go wrong.

Reply:
I have replaced all `int()` to `text.parse_int()`.